client.go

```diff
@@ -250,6 +256,10 @@ func (r *marathonClient) apiCall(method, uri string, body, result interface{}) e
 	request.Header.Add("Content-Type", "application/json")
 	request.Header.Add("Accept", "application/json")
 
+	if r.config.MarathonToken != "" {
+		request.Header.Add("Authorization", "token=" + r.config.MarathonToken)
```

Comment:
This will overwrite any basic authorization set before. While I suppose it doesn't make sense to set both the token and basic auth credentials at the same time, we should leave a comment next to the token field about this circumstance and also have a test for this.